### ISSUE TITLE
When I tried to put the output results into Heartpy, the heart rates I got were completely inconsistent

### ISSUE THREAD

**Issue by jungyin** [ubicomplab/rPPG-Toolbox#319]
I am trying to inject the model output by rppg toolbox into Heartpy to obtain a new result, but it seems completely unsatisfactory as the heart rate calculated by the model fluctuates greatly. Is there any good solution to this?

**Comment by 3051485652** [comment]
> 我正在尝试将 rppg 工具箱输出的模型注入到 Heartpy 中以获得新的结果，但由于模型计算的心率波动很大，似乎完全不令人满意。有什么好的解决方案吗？

哥，您用过里面的无监督学习方法没，能交流一下吗

**Comment by jungyin** [comment]
> > 我正在尝试将 rppg 工具箱输出的模型注入到 Heartpy 中以获得新的结果，但由于模型计算的心率波动很大，似乎完全不令人满意。有什么好的解决方案吗？
> 
> 哥，您用过里面的无监督学习方法没，能交流一下吗

我试了一下，omit那个是比较新一点的，但似乎准确率不是特别理想，看论文和代码，总感觉怪怪的对不太上

**Comment by 3051485652** [comment]
> > > 我正在尝试将 rppg 工具箱输出的模型注入到 Heartpy 中以获得新的结果，但由于模型计算的心率波动很大，似乎完全不令人满意。有什么好的解决方案吗？
> > 
> > 
> > 哥，您用过里面的无监督学习方法没，能交流一下吗
> 
> 我试了一下，omit那个是比较新一点的，但似乎准确率不是特别理想，看论文和代码，总感觉怪怪的对不太上

我也用感觉准确率不高啊，其他无监督POS算法靠谱一点吗？

**Comment by 3051485652** [comment]
> > > 我正在尝试将 rppg 工具箱输出的模型注入到 Heartpy 中以获得新的结果，但由于模型计算的心率波动很大，似乎完全不令人满意。有什么好的解决方案吗？
> > 
> > 
> > 哥，您用过里面的无监督学习方法没，能交流一下吗
> 
> 我试了一下，omit那个是比较新一点的，但似乎准确率不是特别理想，看论文和代码，总感觉怪怪的对不太上

这个方向感觉很多人做啊，怎么找不到交流的人

**Comment by jungyin** [comment]
> > > > 我正在尝试将 rppg 工具箱输出的模型注入到 Heartpy 中以获得新的结果，但由于模型计算的心率波动很大，似乎完全不令人满意。有什么好的解决方案吗？
> > > 
> > > 
> > > 哥，您用过里面的无监督学习方法没，能交流一下吗
> > 
> > 
> > 我试了一下，omit那个是比较新一点的，但似乎准确率不是特别理想，看论文和代码，总感觉怪怪的对不太上
> 
> 这个方向感觉很多人做啊，怎么找不到交流的人

你的精度咋样？现在我搞了俩个星期，也是撞墙了。我用的监督算法，心率跳动很大，真的绷不住了快。

**Comment by 3051485652** [comment]
> > > > > 我正在尝试将 rppg 工具箱输出的模型注入到 Heartpy 中以获得新的结果，但由于模型计算的心率波动很大，似乎完全不令人满意。有什么好的解决方案吗？
> > > > 
> > > > 
> > > > 哥，您用过里面的无监督学习方法没，能交流一下吗
> > > 
> > > 
> > > 我试了一下，omit那个是比较新一点的，但似乎准确率不是特别理想，看论文和代码，总感觉怪怪的对不太上
> > 
> > 
> > 这个方向感觉很多人做啊，怎么找不到交流的人
> 
> 你的精度咋样？现在我搞了俩个星期，也是撞墙了。我用的监督算法，心率跳动很大，真的绷不住了快。

我也是复现了这个omit算法感觉精度不是很高，你

**Comment by yahskapar** [comment]
Hi @jungyin,

Can you please provide more information as to what's going on (e.g., what data you're using, what model you're using, what HeartPy in particular is and whether or not there might be some pre or post processing differences that should be carefully considered)?

**Comment by jungyin** [comment]
I tried using EFFICIONTPHYS and PHYSFormer. Their multiple versions of PTH weights have also been attempted
These are my codes


` if(i1>30 and faces[0].shape[1]>0 and faces[0].shape[0]>0 ):
        dframe = plot_detections(frame,faces[0])
        # dframe = frame
        
        if(len(dframes)<maxfraemlen):
            dframes.append(cv2.resize(dframe,(widthheight,widthheight)))
            dframes2.append(cv2.resize(dframe,(72,72)))
        if len(dframes)==maxfraemlen :
            with torch.no_grad():
            # if False:
                dfs = diff_normalize_data(np.array(dframes,dtype=np.float32))
                dout = np.array(dfs)
                dout=np.transpose(dout, (3,0, 1, 2))
                dout=torch.from_numpy(np.array([dout],dtype=np.float32)).cuda()
                rPPG,_,_,_= PHYSFORMER(dout,2.0)

               
                # np.save('PHYSFORMER.npy',rPPG.cpu().detach().numpy())
            
            # with torch.no_grad():
            if False:
                dfs = diff_normalize_data(np.array(dframes,dtype=np.float32))
                dout = np.array(dfs,dtype=np.float32)
                dout=np.transpose(dout, (3,0, 1, 2))
                dout=torch.from_numpy(dout).cuda()
                rPPG = TSCAN(dout)
                # rPPg1 = OMIT(dout.cpu().detach().numpy())
                # rPPg = torch.transpose(rPPg,1,0)
                # rPPg = rPPg.cpu().detach().numpy()
                # rPPg=np.transpose(rPPg, (1,0))
                
            def workRppg(rPPG):
                # rPPG = rPPG 
                rPPG = (rPPG-torch.mean(rPPG, axis=-1).view(-1, 1))/torch.std(rPPG, axis=-1).view(-1, 1) 
                rPPG = rPPG.detach().cpu().numpy()
                [b, a] = butter(1, [0.75 / fs * 2, 2.5 / fs * 2], btype='bandpass')
            
                rPPG =rPPG[0]
                rPPG =  _detrend(np.cumsum(rPPG),100)
                rPPG1 = scipy.signal.filtfilt(b, a, np.double(rPPG))
                return rPPG1,rPPG
          
            rPPG1,rPPG = workRppg(rPPG)

            num_samples_target = int (len(rPPG1) * 500 / 25)
            hr_value= _calculate_fft_hr(rPPG1,fs)
            # rPPG1 = resample(rPPG1,num_samples_target)
            w,m =hp.process(rPPG1,fs,calc_freq=True)
            w1,m1 =hp.process(rPPG,fs,calc_freq=True)
          
            hr_value1= _calculate_fft_hr(rPPG,fs)
            SBP = DBP = 0
            # y1, y2 = bpmodel.predict(np.reshape(rPPG1[:875],[1,-1,1]))  # 模型计算收缩压、舒张压
            # SBP = int(y1.tolist()[0][0])
            # DBP = int(y2.tolist()[0][0])
            # hr_value = m['bpm']
            # compute_xueyang(rPPG1)
            # hr_value = get_hr(rPPG,24)
            # hr_value =  hr1.detach().cpu().numpy()
            print(m['bpm'],hr_value,(m['breathingrate'] * 60),SBP,DBP)
            # hr_value = m['bpm']
            # if(hr_value != np.nan):
            hps.append(hr_value)`

**Comment by yahskapar** [comment]
Hi @jungyin,

I still have no idea what you are actually trying to do. I'll go ahead and close this issue since it seems more related to HeartPy than this toolbox based on what I translated of the earlier posts, and it's not totally clear to me what you're trying to do with this toolbox.

Feel free to make a new issue if you suspect something is wrong with this toolbox, and you'd like to report a bug, or if have some **specific** question on what you can change to perhaps get a predicted waveform to analyze further using something like HeartPy.
